packages/graphql-transformers-e2e-tests/src/IAMHelper.ts

```diff
@@ -163,12 +165,57 @@ export class IAMHelper {
     return await this.client.deleteRole({ RoleName: roleName }).promise();
   }
 
-  async detachLambdaExecutionPolicy(policyArn: string, roleName: string) {
+  async detachPolicy(policyArn: string, roleName: string) {
     return await this.client
       .detachRolePolicy({
         PolicyArn: policyArn,
         RoleName: roleName,
       })
       .promise();
   }
+
+  async createRole(name: string): Promise<IAM.Role> {
+    const accountDetails = await this.sts.getCallerIdentity({}).promise();
+    const currentAccountId = accountDetails.Account;
+    const role = await this.client
+      .createRole({
+        RoleName: name,
+        AssumeRolePolicyDocument: `{
+          "Version": "2012-10-17",
+          "Statement": [
+            {
+              "Effect": "Allow",
+              "Action": "sts:AssumeRole",
+              "Principal": {
+                "AWS": "${currentAccountId}"
+              },
+              "Condition": {}
+            }
+          ]
+        }`,
+      })
+      .promise();
+    return role.Role;
+  }
+
+  async createAppSyncDataPolicy(policyName: string, region: string, appsyncApiIds: Array<string>) {
+    const accountDetails = await this.sts.getCallerIdentity({}).promise();
+    const currentAccountId = accountDetails.Account;
+    const policyStatement = {
+      Version: '2012-10-17',
+      Statement: [
+        {
+          Effect: 'Allow',
+          Action: ['appsync:GraphQL'],
+          Resource: appsyncApiIds.map((appsyncApiId) => `arn:aws:appsync:${region}:${currentAccountId}:apis/${appsyncApiId}/*`),
```

Comment:
Fair point. The case I'm specifically concerned about is a regression where an IAM control we accidentally apply fine-grained auth rules _in addition to_ the IAM rules. It may be that this case is covered by existing tests, but the schema I remember from this PR are pretty coarse grained, and the case I'm wondering about is a schema with field-level auth:

```graphql
type Todo @model @auth(rules: [
{ allow: owner },
{ allow: public, operations: [read] },
]) {
  id: ID!
  name: String!
  description: String
  secret: String @auth(rules: [{allow: owner}])
}
```

Reply:
I can add such test.

Reply:
However, after looking once again. it seems that this is covered by `AuthV2ExhaustiveT3B.test.ts` and `AuthV2ExhaustiveT3D.test.ts` which generate schemas with "partial access" .
I.e. I copied existing `AuthV2ExhaustiveT2...` tests that cover `iam` provider in auth directive and run them with iam access setting enabled. They assert that for cognito traffic directive works as before.

Reply:
also added tests in 
https://github.com/aws-amplify/amplify-category-api/pull/2331/commits/ca778e9be3a204d96b8407e0d22c9bbd580e67b5
and 
https://github.com/aws-amplify/amplify-category-api/pull/2331/commits/8602e341692633ab65c106d9abed31072a2df297